types/selenium-webdriver/lib/error.d.ts

```diff
@@ -0,0 +1,344 @@
+ * @param {string=} opt_error the error message, if any.
+ */
+export class StaleElementReferenceError extends WebDriverError {
+    constructor(opt_error?: string);
+}
+
+/**
+ * An operation did not complete before its timeout expired.
+ * @param {string=} opt_error the error message, if any.
+ */
+export class TimeoutError extends WebDriverError {
+    constructor(opt_error?: string);
+}
+
+/**
+ * A request to set a cookie’s value could not be satisfied.
+ * @param {string=} opt_error the error message, if any.
+ */
+export class UnableToSetCookieError extends WebDriverError {
+    constructor(opt_error?: string);
+}
+
+/**
+ * A screen capture operation was not possible.
+ * @param {string=} opt_error the error message, if any.
+ */
+export class UnableToCaptureScreenError extends WebDriverError {
+    constructor(opt_error?: string);
+}
+
+/**
+ * A modal dialog was open, blocking this operation.
+ * @param {string=} opt_error the error message, if any.
+ * @param {string=} opt_text the text of the open dialog, if available.
+ */
+export class UnexpectedAlertOpenError extends WebDriverError {
+    constructor(opt_error?: string, opt_text?: string);
+
+    /**
+     * @return {(string|undefined)} The text displayed with the unhandled alert,
+     *     if available.
+     */
+    getAlertText(): string | undefined;
+}
+
+/**
+ * A command could not be executed because the remote end is not aware of it.
+ * @param {string=} opt_error the error message, if any.
+ */
+export class UnknownCommandError extends WebDriverError {
+    constructor(opt_error?: string);
+}
+
+/**
+ * The requested command matched a known URL but did not match an method for
+ * that URL.
+ * @param {string=} opt_error the error message, if any.
+ */
+export class UnknownMethodError extends WebDriverError {
+    constructor(opt_error?: string);
+}
+
+/**
+ * Reports an unsupported operation.
+ * @param {string=} opt_error the error message, if any.
+ */
+export class UnsupportedOperationError extends WebDriverError {
+    constructor(opt_error?: string);
+}
+
+/**
+ * Checks a legacy response from the Selenium 2.0 wire protocol for an error.
+ * @param {*} responseObj the response object to check.
+ * @return {*} responseObj the original response if it does not define an error.
+ * @throws {WebDriverError} if the response object defines an error.
+ */
+export function checkLegacyResponse<T>(responseObj: T): T;
+
+/**
+ * @param {*} err The error to encode.
+ * @return {{error: string, message: string}} the encoded error.
+ */
+export function encodeError(err: any): { error: string; message: string };
+
+/**
+ * Tests if the given value is a valid error response object according to the
+ * W3C WebDriver spec.
+ *
+ * @param {?} data The value to test.
+ * @return {boolean} Whether the given value data object is a valid error
+ *     response.
+ * @see https://w3c.github.io/webdriver/webdriver-spec.html#protocol
+ */
+export function isErrorResponse(data?: any): boolean;
```

Comment:
This could _maybe_ be a [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates), but it's not clear to me from reading the spec what `data` looks like if it's true. So `boolean` return type is probably fine.

Reply:
Not sure if a type guard is appropriate, since the function only verifies if data is an object and contains the property error with the type string. I assume that it'd be used for the verification of if the obj can be used with the function `throwDecodedError` or not, but since it's not explicitly checked for the property message, I didn't want to enforce the type to be strictly `{error:string}`. That would mean that message would be potentially missing from the type.
However, if you think a type guard would be appropriate here, we could use the following: 
```
export function isErrorResponse(data?: any): data is {error: string};
```

Reply:
I think it's fine either way. It looks like with expected usage of the function, the type guard doesn't provide a ton of value.